js/src/edit-block/hooks/useBlock.js

```diff
@@ -33,21 +35,58 @@ const useBlock = () => {
 	);
 
 	const fullBlock = getFullBlock();
-	const blockNameWithNamespace = Object.keys( fullBlock )[ 0 ];
-	const block = fullBlock[ blockNameWithNamespace ];
+
+	const blockNameWithNameSpace = getBlockNameWithNameSpace( fullBlock );
+	const block = useMemo(
+		() => fullBlock[ blockNameWithNameSpace ] || {},
+		[ fullBlock, blockNameWithNameSpace ]
```

Comment:
This way, it will only calculate `block` when `fullBlock` or `blockNameWithNameSpace` change.